core/cas-server-core-authentication-mfa-api/src/main/java/org/apereo/cas/authentication/DefaultRequestedAuthenticationContextValidator.java

```diff
@@ -37,6 +39,26 @@
             return Pair.of(Boolean.TRUE, Optional.empty());
         }
 
-        return authenticationContextValidator.validate(authentication, requestedContext.get(), registeredService);
+        val providerId = requestedContext.get();
```

Comment:
Where does this come from?

Reply:
From multifactorTriggerSelectionStrategy.resolve().  If a MFA provider is determined by the selection strategy triggers. then we need to make that it isn't explicitly by passed for the request and that the provider is available.